components/Terminal.js

```diff
@@ -43,24 +61,20 @@ class Terminal extends Component {
     prompt: PropTypes.string,
     barColor: PropTypes.string,
     backgroundColor: PropTypes.string,
-    commands: PropTypes.objectOf(PropTypes.oneOfType([
-      PropTypes.func,
-      PropTypes.shape({
-        options: PropTypes.arrayOf(PropTypes.shape({
-          name: PropTypes.string,
-          description: PropTypes.string,
-          defaultValue: PropTypes.any,
-        })),
-        method: PropTypes.func,
-      }),
-    ])),
-    description: PropTypes.objectOf(PropTypes.string),
+    commands: commandsPropType,
+    descriptions: descriptionsPropType,
     watchConsoleLogging: PropTypes.bool,
     commandPassThrough: PropTypes.oneOfType([
       PropTypes.func,
       PropTypes.bool,
     ]),
     promptSymbol: PropTypes.string,
+    plugins: PropTypes.arrayOf(PropTypes.shape({
```

Comment:
Its great that we have a plugin system but what about the docs ? about creating a one ?